src/Assets/Javascript/linkedList.js

```diff
@@ -89,84 +92,111 @@ export default class LinkedList {
     allArray.push(currentNode.value);
     while (currentNode.nextNode) {
       currentNode = currentNode.nextNode;
       allArray.push(currentNode.value);
     }
-
-    return allArray;
+    const newArray = [];
+    let stage = 0;
+    for (let i = 0; i < allArray.length; i += 1) {
+      while (allArray[i].index === stage + 1) {
+        newArray.push(allArray[stage]);
+        stage += 1;
+      }
+    }
+    return newArray;
   }
 
-  remove(index = null) {
-    if (index) {
-      if (this.head) {
-        let valArray = [];
-        let activeValue = this.head;
-        valArray.push(activeValue.value);
-        while (activeValue.nextNode) {
-          activeValue = activeValue.nextNode;
-          valArray.push(activeValue.value);
-        }
-        if (index > valArray.length) {
-          return false;
-        }
-        const newArray = [];
-        for (let i = 0; i < valArray.length; i += 1) {
-          if (i !== index) {
-            newArray.push(valArray[i]);
-          }
-        }
-        const returnValue = valArray[index];
-        valArray = newArray.reverse();
-        if (newArray.length !== 0) {
-          let onHold = new NewNode(valArray[0]);
-          for (let i = 1; i < valArray.length; i += 1) {
-            onHold = new NewNode(valArray[i], onHold);
-          }
-          this.head = onHold;
-        } else {
-          this.head = null;
+  changeState(value, bool) {
+    if (this.head) {
+      let element = this.head;
+      while (element.nextNode) {
+        if (element.value.index === value) {
+          element.value.completed = bool;
         }
-        return returnValue;
+        element = element.nextNode;
       }
-      return false;
-    }
-    if (!this.head) {
-      return -1;
+      return true;
     }
-    let valArray = [];
-    let activeValue = this.head;
+    return false;
+  }
 
-    valArray.push(activeValue.value);
-    while (activeValue.nextNode) {
-      activeValue = activeValue.nextNode;
-      valArray.push(activeValue.value);
+  /*                                                ON DEVELOPMENT
+  removeState(){
+    let tempList = this.head
+    let elemArray = []
+    while(tempList.value){
+      if(tempList.value.completed === false){
+        let tempNode = new NewNode(tempList.value)
+        elemArray.push(tempNode)
+        tempList = tempList.nextNode
+      }
+      else{
+        tempList = tempList.nextNode
+      }
     }
-    const subsArray = [];
-    for (let i = 0; i < valArray.length; i += 1) {
-      if (i !== valArray.length - 1) {
-        subsArray.push(valArray[i]);
+    let newLinkedList = new NewNode(elemArray[elemArray.length-1])
+    newLinkedList.value.index = 5
+    for(let i = 1;i<elemArray.length;i+=1){
+        const newItem = new NewNode(elemArray[elemArray.length-(i+1)], newLinkedList);
+     newItem.value.index = elemArray.length-(i)
+        newLinkedList = newItem
+    }
+    console.log(newLinkedList)
+    this.head = newLinkedList
+
+  }
```

Comment:
[Optional]
You should not commit commented-out code. It will make your project look unprofessional.